### ISSUE TITLE
Getting PAGE_TRANSPORT_TIMEOUT for any query that goes beyond 1 minute.

### ISSUE THREAD

**Issue by ak2766** [trinodb/trino#24149]
TL;DR - Which timeout setting do I need to change from default to stop this PAGE_TRANSPORT_TIMEOUT and does it need to go in coordinator, worker, or both?

I'm experiment with Trino and DBeaver and I'm hitting a road block for queries that take longer than 1 minute to complete.

For instance, the query below in particular never finishes when using Trino cli.  However, if I run the same query on SMS, it completes in ~3 minutes.  The table is >30GB

```
trino> SELECT id, REDACTED, REDACTED, REDACTED, count, REDACTED 
    -> from mssql.dbo.REDACTED 
    -> order by count desc
    -> limit 200;

Query 20241117_033406_00006_u26jx, FAILED, 1 node
Splits: 1 total, 0 done (0.00%)
1:07 [1 rows, 0B] [0 rows/s, 0B/s]

Query 20241117_033406_00006_u26jx failed: Encountered too many errors talking to a worker node. The node may have crashed or be under too much load. This is probably a transient issue, so please retry your query in a few minutes. (http://REDACTED:8080/v1/task/20241117_033406_00006_u26jx.0.0.0/results/0/0 - 104 failures, failure duration 60.02s, total failed request time 61.74s)
```

Initially, thought it was a DBeaver issue and I wasted hours researching timeouts.  After exhausting all timeouts on DBeaver, I finally tried running the query directly on Trino's (which I really ought to have done first), I discovered the timeout is occurring somewhere inside Trino.  I just started on Trino yesterday so I'm very green.

Here are my configs:

<details>
<summary>coordinator config:</summary>

```
coordinator=true
node-scheduler.include-coordinator=false
http-server.http.port=8080
discovery-server.enabled=true
discovery.uri=http://trino-coordinator:8080
internal-communication.shared-secret=REDACTED
internal-communication.https.required=false
```
</details>

<details>
<summary>worker config:</summary>

```
coordinator=false
http-server.http.port=8080
discovery.uri=http://trino-coordinator:8080
internal-communication.shared-secret=REDACTED
internal-communication.https.required=false
```
</details>

**Comment by zachtrong** [comment]
There are undocumented `http-client` config properties to increase timeout:

```yaml
  workerExtraConfig: |-
    exchange.http-client.request-timeout=60s
    exchange.http-client.idle-timeout=2m 
    exchange.http-client.max-connections-per-server=1000
  coordinatorExtraConfig: |-
    exchange.http-client.request-timeout=60s
    exchange.http-client.idle-timeout=2m 
    exchange.http-client.max-connections-per-server=1000
```

**Comment by ak2766** [comment]
Thanks @zachtron.

I went through the logs and searched for timeouts.  Trial and error got me to the correct one and eventually got it going a day ago but forgot to come back and update.

**Comment by dirksan28** [comment]
@ak2766 I'm experiencing similar issues and have been searching endlessly for a solid starting point to tweak things. Could you share which timeout setting made a noticeable difference for you?

**Comment by ak2766** [comment]
Dang it - I wonder why I did not put in that useful information.

[rant to self]
I really hate it when people post back that they got things working without providing details on what they did to resolve.  Yet, here I am doing the EXACT SAME THING - tsk, tsk.
[/rant]


PS: Let me hunt down all my current settings and post back for posterity.  Give me ~24hrs.

**Comment by ak2766** [comment]
This is my current config that works well with large datasets:
<details>
<summary>coordinator config:</summary>

```
coordinator=true
node-scheduler.include-coordinator=false
http-server.http.port=8080
discovery-server.enabled=true
discovery.uri=http://trino-coordinator:8080
internal-communication.shared-secret=Sup3Rs3cret!
internal-communication.https.required=false
query.remote-task.max-error-duration=10m
exchange.http-client.max-content-length=4GB
query.remote-task.max-request-size=10GB
query.max-memory-per-node=6GB
```
</details>

<details>
<summary>worker config:</summary>

```
coordinator=false
http-server.http.port=8080
discovery.uri=http://trino-coordinator:8080
internal-communication.shared-secret=Sup3Rs3cret!
internal-communication.https.required=false
query.remote-task.max-error-duration=10m
sink.max-buffer-size=5120MB
exchange.max-buffer-size=5120MB
exchange.deduplication-buffer-size=5120MB
driver.max-page-partitioning-buffer-size=5120MB
exchange.http-client.max-content-length=4GB
query.remote-task.max-request-size=10GB
query.max-memory-per-node=6GB
```
</details>

If I remember correctly, the key - `exchange.http-client.max-content-length=4GB` - was actually what resolved the issue for me.

Let me know if that resolves your issue as well.

**Comment by ak2766** [comment]
I'll go ahead and close this one now that I'm here.

**Comment by judnich** [comment]
We also ran into this issue a few times, so I wouldn't necessarily consider this issue fixed in main -- at least, it's not clear what the real root cause is, otherwise there probably should be changes to the default timeouts or something.

For the benefit of others encountering this issue, I also seem to have resolved it by just bumping up a bunch of timeouts and buffer sizes, thanks to inspiration from the comments in this github issue. Though this is mostly a shot in the dark that just seems to have worked -- it is still slightly unsettling not knowing the true root cause.

I dug through the Trino source a bit and also found a number of undocumented (and some documented) properties governing inter-worker (and also coordinator) timeouts and buffer sizes. Then, I approximately doubled most of the relevant buffers, and significantly bumped up the timeouts (especially those that were just a few seconds):

```
query.client.timeout=60m
query.remote-task.max-error-duration=10m
query.remote-task.max-request-size=32MB

exchange.http-client.max-content-length=64MB
exchange.http-client.idle-timeout=60s
exchange.http-client.request-timeout=30s
exchange.max-error-duration=5m
exchange.max-buffer-size: 64MB
exchange.max-response-size: 32MB
exchange.deduplication-buffer-size=64MB

sink.max-buffer-size: 64MB
sink.max-broadcast-buffer-size: 512MB
```
